rust/connlib/snownet/src/node.rs

```diff
@@ -181,6 +210,65 @@ where
         self.connections.len()
     }
 
+    /// Upserts a connection to the given remote.
+    ///
+    /// If we already have a connection with the same ICE credentials, this does nothing.
+    /// Otherwise, the existing connection is discarded and a new one will be created.
+    #[tracing::instrument(level = "info", skip_all, fields(%cid))]
+    pub fn upsert_connection(
```

Comment:
Should this be exposed by the gateway/client in this PR?

Reply:
No, it will start to be used in #6941 and #6942.